bioio/plugins.py

```diff
@@ -233,6 +233,11 @@ def get_plugins(use_cache: bool) -> Dict[str, List[PluginEntry]]:
     # Save copy of plugins to cache then return
     plugins_by_ext_cache.clear()
     plugins_by_ext_cache.update(plugins_by_ext)
+
+    # Sort dict from longest to shortest key
+    plugins_by_ext = dict(
+        sorted(plugins_by_ext.items(), key=lambda x: len(x[0]), reverse=True)
+    )
```

Comment:
A few concerns about this.
1. you're changing the plugins_by_ext after its previous value was stored in the plugins_by_ext_cache.
2. Is this completely undoing the "insert in sorted order by most recently installed" which we did above?
3. I'm not sure I understand how the sorting is happening. Is this a global sort over all extensions for all installed plugins? If so then you're not just prioritizing ometiff over tiff, but really scrambling the order.

Reply:
Completely agree with many points. This was my hacky fix before I needed to go before the end of the day and should have made this a draft PR.

Iirc our original goal was to sort the dict keys (file extensions) by longest first, and then within  each dict value (the list of plugin entries) sort the list of plugins by their install time.

The goal was basically both "always choose the most specific suffix and then within a suffix that has multiple supported readers, choose the most recently installed one"

I'll make changes tomorrow

Reply:
I can also do this processing at `determine_plugin` time if that's better?